sn_transfers/src/wallet/local_store.rs

```diff
@@ -299,111 +302,108 @@ impl LocalWallet {
-                    .ok_or(WalletError::TotalPriceTooHigh)?;
-
-                unique_key_vec.push((
-                    amount,
-                    address,
-                    UniquePubkey::random_derivation_index(&mut rng),
-                ));
-            }
-
-            // add network royalties payment as payee as well
-            let royalties = calculate_royalties_fee(cost);
-            payees.push((*NETWORK_ROYALTIES_PK, royalties));
-            unique_key_vec.push((
-                royalties,
+        // create random derivation indexes for recipients
+        let mut recipients_by_xor = BTreeMap::new();
+        for (xorname, (main_pubkey, cost)) in price_map.iter() {
+            let storage_payee = (
+                *cost,
+                *main_pubkey,
+                UniquePubkey::random_derivation_index(&mut rng),
+            );
+            let royalties_fee = calculate_royalties_fee(*cost);
+            let royalties_payee = (
+                royalties_fee,
                 *NETWORK_ROYALTIES_PK,
                 UniquePubkey::random_derivation_index(&mut rng),
-            ));
-
-            all_payees_only.extend(unique_key_vec);
+            );
 
             storage_cost = storage_cost
-                .checked_add(cost)
+                .checked_add(*cost)
                 .ok_or(WalletError::TotalPriceTooHigh)?;
             royalties_fees = royalties_fees
-                .checked_add(royalties)
+                .checked_add(royalties_fee)
                 .ok_or(WalletError::TotalPriceTooHigh)?;
-        }
 
-        let reason_hash = reason_hash.unwrap_or_default();
+            recipients_by_xor.insert(xorname, (storage_payee, royalties_payee));
+        }
 
+        // create offline transfers
+        let recipients = recipients_by_xor
+            .values()
+            .flat_map(|(node, roy)| vec![node, roy])
+            .cloned()
+            .collect();
         let (available_cash_notes, exclusive_access) = self.available_cash_notes()?;
         debug!("Available CashNotes: {:#?}", available_cash_notes);
+        let reason_hash = Default::default();
         let offline_transfer = create_offline_transfer(
             available_cash_notes,
-            all_payees_only,
+            recipients,
             self.address(),
             reason_hash,
         )?;
 
-        let mut all_transfers_per_address = BTreeMap::default();
-
-        let mut used_cash_notes = std::collections::HashSet::new();
-
-        for (content_addr, payees) in all_data_payments {
-            for (payee, token) in payees {
-                if let Some(cash_note) =
-                    offline_transfer
-                        .created_cash_notes
-                        .iter()
-                        .find(|cash_note| {
-                            cash_note.main_pubkey() == &payee
-                                && !used_cash_notes.contains(&cash_note.unique_pubkey().to_bytes())
-                        })
```

Comment:
This code here checks only the `main_pubkey`, so if one node has two chunks payements for them in this batch we might mix the outputs!

Say we have two records of two different prices somehow:
chunk1, price1
chunk2, price2

What prevents us from using the wrong one?